Fix `NoneType` error when querying SOA records

maxkrivich opened the pull request:
Hey there 👋🏻 

In this PR, I have made changes to `fierce` function to fix `NoneType` error whenever SOA query return `None`. After applying the patch the tool will gradually shutdown and print a proper error message.  I have tested the changes locally, and they work perfectly.

Example of the error:
```bash
fierce --domain belgiantrain.be

NS: ns2.belgiantrain.be. ns1.belgiantrain.be.
Traceback (most recent call last):
  File "/Users/user/Projects/fierce/.venv/bin/fierce", line 33, in <module>
    sys.exit(load_entry_point('fierce', 'console_scripts', 'fierce')())
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/Projects/fierce/fierce/fierce.py", line 491, in main
    fierce(**vars(args))
  File "/Users/user/Projects/fierce/fierce/fierce.py", line 322, in fierce
    master_address = master[0].address
                     ~~~~~~^^^
TypeError: 'NoneType' object is not subscriptable
```

I would love to hear your feedback on these changes. Please let me know if there's anything you'd like me to modify or improve. I'm here to make any necessary adjustments to ensure this contribution aligns with the project's standards.

Cheers,
Max 👨🏻‍💻

mschwager reviewed (approved):
Nice find! And thank you for the fix and comprehensive description :+1:

maxkrivich commented:
@mschwager I've fixed the error with Github actions by removing old versions of Python that are no longer available. Now it should pass the checks.

mschwager commented:
> @mschwager I've fixed the error with Github actions by removing old versions of Python that are no longer available. Now it should pass the checks.

Nice, thank you. If you're going to drop old Python version support in this PR, can you update these locations too:

- https://github.com/mschwager/fierce/blob/master/setup.py#L41
- And add a `Removed` section: https://github.com/mschwager/fierce/blob/master/CHANGELOG.md#unreleased

maxkrivich commented:
> > @mschwager I've fixed the error with Github actions by removing old versions of Python that are no longer available. Now it should pass the checks.
> 
> Nice, thank you. If you're going to drop old Python version support in this PR, can you update these locations too:
> 
>     * [`master`/setup.py#L41](https://github.com/mschwager/fierce/blob/master/setup.py?rgh-link-date=2023-07-29T15%3A50%3A16Z#L41)
> 
>     * And add a `Removed` section: [`master`/CHANGELOG.md#unreleased](https://github.com/mschwager/fierce/blob/master/CHANGELOG.md?rgh-link-date=2023-07-29T15%3A50%3A16Z#unreleased)

Good catch. I've adjusted the PR according to the comment. Moreover, I decided to pin down the minimal required Python version to 3.8. It will reduce the maintenance burden for cases if someone still uses the deprecated version of Python.

mschwager commented on CHANGELOG.md:
```diff
@@ -1,98 +1,145 @@
 ## [Unreleased]
 
+## [1.5.1] - 2023-07-30
+
```
```suggestion
```

mschwager commented:
Thanks for fixing the tests! I dropped the `1.5.1` changes from this PR. We can add that back in when it makes sense to release a new version :+1:
